### PR TITLE
Initial model for data sharing page texts

### PR DESCRIPTION
**Pull request by ziafazal**
**Description:** Model to persist data sharing consent page texts

**JIRA:** https://openedx.atlassian.net/browse/WL-1246

### CODE REVIEW

File: consent/models.py [hunk 1]
```diff
@@ -239,3 +246,150 @@ class Meta(Consent.Meta):
...
+        help_text=_('Header text in left sidebar. platform_name variable may be available')
+    )
+    left_sidebar_text = models.TextField(
+        null=True,
+        blank=True,
+        default=_(
+            "{enterprise_customer_name} has partnered with {platform_name} to offer you high-quality learning "
+            "opportunities from the world's best universities."
+        ),
+        help_text=SIDEBAR_PARAGRAPH_HELP_TEXT
```
Comment (bill-filler) [inline consent/models.py]: wondering if we could combine left_sidebar_header and left_sidebar_text into one field: left_sidebar_paragraph. To achieve the formatting that is in the default template the paragraph text would need to have markup built in to get <p> between the header and the text

File: consent/models.py [hunk 2]
```diff
@@ -239,3 +246,150 @@ class Meta(Consent.Meta):
...
+        blank=True,
+        default=_('Consent to share your data'),
+        help_text=_('Heading of page')
+    )
+    top_paragraph = models.TextField(
+        null=True,
+        blank=True,
+        default=_('First paragraph of page'),
+        help_text=TOP_PARAGRAPH_HELP_TEXT
+    )
```
Comment (bill-filler) [inline consent/models.py]: same comment about message_header and top_paragraph, maybe combine into one field?

File: consent/models.py [hunk 3]
```diff
@@ -239,3 +246,150 @@ class Meta(Consent.Meta):
...
+        default=_('No, take me back.'),
+        help_text=_('Text of decline link')
+    )
+    policy_dropdown_header = models.CharField(
+        max_length=255,
+        null=True,
+        blank=True,
+        default=_('Data Sharing Policy'),
+        help_text=_('Text of policy drop down')
+    )
```
Comment (bill-filler) [inline consent/models.py]: We might needy another field like show_policy_details that is a boolean. If true, then we allow displaying policy_dropdown_header and policy_paragraph and if false we do not show those fields. We need this for the Boeing/MIT use case where they do not show the policy dropdown or text at all (see attached screenshot)
boeingdsc

File: consent/models.py [hunk 4]
```diff
@@ -239,3 +246,150 @@ class Meta(Consent.Meta):
...
+        max_length=255,
+        default=_('View the data sharing policy'),
+        help_text=_('Text of abort decline link on confirmation dialog box')
+    )
+    enterprise_customer = models.OneToOneField(
+        EnterpriseCustomer,
+        related_name="enterprise_consent_page",
+        on_delete=models.deletion.CASCADE
+    )
+    history = HistoricalRecords()
```
Comment (bill-filler) [inline consent/models.py]: is history necessary? not sure, maybe @douglashall has an opinion

File: consent/models.py (hunk 3 above)
Comment (ziafazal) [reply consent/models.py]: I was thinking of displaying policy section only if both `policy_dropdown_header ` and `policy_paragraph` are not empty.

**Comment by johnnagro** [comment]
I am closing old, inactive PRs. If you feel this has been closed in error please comment.

**Pull request closed without merging**